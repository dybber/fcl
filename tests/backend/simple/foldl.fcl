sig runtestBlock : [int] -> Program <block> [int]<block>
fun runtestBlock arr =
  splitUp 32 arr
   |> mapPull (foldl addi 0)
   |> mapPull (return <thread>)
   |> concat 1

sig runtest : int -> Program <grid> [int]<grid>
fun runtest n =
  splitUp n (iota n)
   |> mapPull runtestBlock
   |> concat 32

sig main : Program <grid> [int]
fun main =
 do { a <- runtest (32*32)
    ; forceAndPrint 32 a
    }
