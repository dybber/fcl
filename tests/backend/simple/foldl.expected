[496,1520,2544,3568,4592,5616,6640,7664,8688,9712,10736,11760,12784,13808,14832,15856,16880,17904,18928,19952,20976,22000,23024,24048,25072,26096,27120,28144,29168,30192,31216,32240]
